parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -470,27 +473,50 @@ private void checkAutoCommitIsDisabled(org.apache.kafka.clients.consumer.Consume
         try {
             if (consumer instanceof KafkaConsumer) {
                 // Could use Commons Lang FieldUtils#readField - but, avoid needing commons lang
-                Field coordinatorField = KafkaConsumer.class.getDeclaredField("coordinator");
-                coordinatorField.setAccessible(true);
-                ConsumerCoordinator coordinator = (ConsumerCoordinator) coordinatorField.get(consumer); //IllegalAccessException
-
-                if (coordinator == null)
-                    throw new IllegalStateException("Coordinator for Consumer is null - missing GroupId? Reflection broken?");
-
-                Field autoCommitEnabledField = coordinator.getClass().getDeclaredField("autoCommitEnabled");
-                autoCommitEnabledField.setAccessible(true);
-                Boolean isAutoCommitEnabled = (Boolean) autoCommitEnabledField.get(coordinator);
+                Field delegateField = KafkaConsumer.class.getDeclaredField("delegate");
+                delegateField.setAccessible(true);
+                ConsumerDelegate<?, ?> delegate = (ConsumerDelegate<?, ?>) delegateField.get(consumer);
+
+                Boolean isAutoCommitEnabled;
+                if (delegate instanceof LegacyKafkaConsumer) {
+                    Field coordinatorField = LegacyKafkaConsumer.class.getDeclaredField("coordinator");
+                    coordinatorField.setAccessible(true);
+                    ConsumerCoordinator coordinator = (ConsumerCoordinator) coordinatorField.get(delegate); //IllegalAccessException
+
+                    if (coordinator == null)
+                        throw new IllegalAccessException("Coordinator for Consumer is null - missing GroupId? Reflection broken?");
+
+                    Field autoCommitEnabledField = coordinator.getClass().getDeclaredField("autoCommitEnabled");
+                    autoCommitEnabledField.setAccessible(true);
+                    isAutoCommitEnabled = (Boolean) autoCommitEnabledField.get(coordinator);
+                } else if (delegate instanceof AsyncKafkaConsumer) {
+                    Field autoCommitEnabledField = AsyncKafkaConsumer.class.getDeclaredField("autoCommitEnabled");
+                    autoCommitEnabledField.setAccessible(true);
+                    isAutoCommitEnabled = (Boolean) autoCommitEnabledField.get(delegate);
+                } else {
+                    log.error("Consumer delegate is neither a LegacyKafkaConsumer nor a AsyncKafkaConsumer - cannot check auto commit is disabled for consumer type: " + consumer.getClass().getName());
+                    isAutoCommitEnabled = false;
```

Comment:
Check the new option flag and either throw an error or log a warning depending on how that flag is set - something like below:
```suggestion
               if (options.ignoreReflectiveAccessExceptionsForAutoCommitDisabledCheck()) {
                    log.warn("Consumer delegate is neither a LegacyKafkaConsumer nor a AsyncKafkaConsumer - cannot check auto commit is disabled for consumer type: {}. Ignoring due to the option flag ignoreReflectiveAccessExceptionsForAutoCommitDisabledCheck=true", consumer.getClass().getName());
                } else {
                    throw new ParallelConsumerException("Consumer delegate is neither a LegacyKafkaConsumer nor a AsyncKafkaConsumer - cannot check auto commit is disabled for consumer type: " + consumer.getClass().getName());
                }
```